Login acceptance tests, improve login form

sneridagh opened the pull request:
(WIP)

pnicolli commented:
Test is simple and straightforward and I'm ok with it at this time. Looking ahead in the future, I am not a fan of having the tests in a different package from the code, but I'm not a cypress expert, I'm not sure if there's a way that is still simple and allows test code to be colocated with add-on code.

sneridagh commented:
@pnicolli I had exactly the same doubt yesterday. We need to talk about it. I want to add a couple of more tests, and probably implement a logout. It's what testing things has, it brings more things to do!

sneridagh commented:
@pnicolli ready now. I centralized the Cypress testing in `@plone/tooling`, then all the calls are made from there.
In the process, I also fixed some things here and there, removed legacy things like the `i18n` script, removed the deps in there, and other improvements.

stevepiercy commented on apps/seven/news/6944.feature:
```diff
@@ -0,0 +1 @@
+Cypress tests using centralized config in `@plone/tooling`. @sneridagh
```
```suggestion
Modify Cypress tests to use a centralized configuration in `@plone/tooling`. @sneridagh
```

stevepiercy commented on packages/cmsui/news/6944.feature:
```diff
@@ -0,0 +1 @@
+Cypress tests using centralized config in `@plone/tooling`. @sneridagh
```
```suggestion
Modify Cypress tests to use a centralized configuration in `@plone/tooling`. @sneridagh
```

stevepiercy commented on packages/tooling/cypress/support/e2e.js:
```diff
@@ -18,8 +18,9 @@ Cypress.on('uncaught:exception', (err) => {
+
   if (
-    /hydrat/i.test(err.message) ||
+    /ydrat/i.test(err.message) ||
```
Is this an accidental deletion?

Also it looks like a copy-paste from a GitHub issue https://github.com/cypress-io/cypress/issues/27204#issuecomment-1894217901. I think we should have an issue in Volto's issue tracker to track it and remove it when it is fixed. That will also allow you to spread the maintenance workload and relieve your brain of being the sole source.

sneridagh replied:
Not a typo, is fine. Sometimes the error message has a Capital letter (because it's the start of a sentence) and in other occasions, it does not. The error is still present in Cypress 14 unfortunately :(

Anyways, it's a minor thing.

stevepiercy reviewed (changes requested):
Change log tweaks, a question, and a suggestion

sneridagh commented:
@pnicolli going ahead and merging! Please tell me if there's anything that you miss/change!
